android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskExpireWorker.kt

```diff
@@ -40,30 +35,12 @@ constructor(
   @Assisted workerParams: WorkerParameters,
   val fhirEngine: FhirEngine,
   val fhirTaskExpireUtil: FhirTaskExpireUtil,
-  val sharedPreferences: SharedPreferencesHelper,
   val dispatcherProvider: DispatcherProvider
 ) : CoroutineWorker(context, workerParams) {
 
   override suspend fun doWork(): Result {
     return withContext(dispatcherProvider.io()) {
-      val lastAuthoredOnDate =
-        sharedPreferences
-          .read(SharedPreferenceKey.OVERDUE_TASK_LAST_AUTHORED_ON_DATE.name, null)
-          ?.parseDate(SDF_YYYY_MM_DD)
-
-      var (maxDate, tasks) =
-        fhirTaskExpireUtil.expireOverdueTasks(lastAuthoredOnDate = lastAuthoredOnDate)
-
-      while (tasks.isNotEmpty()) {
-        val resultPair = fhirTaskExpireUtil.expireOverdueTasks(lastAuthoredOnDate = maxDate)
-        maxDate = resultPair.first
-        tasks = resultPair.second
-      }
-
-      sharedPreferences.write(
-        SharedPreferenceKey.OVERDUE_TASK_LAST_AUTHORED_ON_DATE.name,
-        maxDate?.formatDate(SDF_YYYY_MM_DD)
-      )
```

Comment:
Why is this code deleted? This was to act as measure so as not to freeze the application when the background job is running.

Reply:
How is this related to the tagged issue?